internal/boxcli/run.go

```diff
@@ -113,3 +115,55 @@ func parseScriptArgs(args []string, flags runCmdFlags) (string, string, []string
 
 	return flags.config.path, script, scriptArgs, nil
 }
+
+func wrapArgsForRun(args []string) []string {
+	for _, arg := range args {
+		if arg == "--" {
+			return args
+		}
+	}
+
+	// if the first argument is not "run", we don't need to do anything. If there
+	// are 2 or fewer arguments, we also don't need to do anything because there
+	// are no flags after a non-run non-flag arg.
+	// IMPROVEMENT: technically users can pass a flag before the subcommand "run"
+	if len(args) <= 2 || args[0] != "run" {
+		return args
+	}
+
+	runFlags := runCmd().Flags()
+	// typical args can be of the form:
+	// run --flag1 val1 -f val2 --flag3=val3 --bool-flag python --version
+	// We handle each different type of flag
+	// (flag with equals, long-form, short-form, and defaulted flags)
+	// Note that defaulted does not mean initial value, it only means flags
+	// that don't require a value.
+	// For example, --bool-flag has NoOptDefVal set to "true".
+	i := 1
+	for i < len(args) {
+		arg := args[i]
+		if strings.HasPrefix(arg, "-") {
+			if strings.Contains(arg, "=") {
+				i++
+				continue
+			}
+			var flag *pflag.Flag
+			if strings.HasPrefix(arg, "--") {
+				flag = runFlags.Lookup(strings.TrimLeft(arg, "-"))
+			} else {
+				flag = runFlags.ShorthandLookup(strings.TrimLeft(arg, "-"))
+			}
+			if flag != nil {
+				if flag.NoOptDefVal == "" {
+					i++
+				}
+				i++
+				continue
+			}
+		}
+
+		return append(args[:i+1], append([]string{"--"}, args[i+1:]...)...)
```

Comment:
I think this logic make sense. Can we simplify a bit? Took me a while to understand what is going on. like:

```
for i < len(args) {
  arg := args[i]
  if !strings.HasPrefix(arg, "-") {
    // finally found it!
    return append(args[:i+1], append([]string{"--"}, args[i+1:]...)...)
  }

  if strings.Contains(arg, "=") {
    i++
    continue
  }
  
  var flag *pflag.Flag
  if strings.HasPrefix(arg, "--") {
    flag = runFlags.Lookup(strings.TrimLeft(arg, "-"))
  } else {
    flag = runFlags.ShorthandLookup(strings.TrimLeft(arg, "-"))
  }
  if flag != nil {
    if flag.NoOptDefVal == "" {
	i := i+2
    } else {
        i++
    }
}
```

Reply:
Yeah. I would also add a comment in english stating what the loop is doing.

Reply:
Something like: "adds a `--` after all flags that come before the command that will be run. Thus, it transforms something like `run --flag1 val1 command --flag2 val2` into `run --flag1 val1 -- command --flag2 val2`".

That would save a code reader quite a bit of time I think.